### PR TITLE
ContextualLexer now uses self.basic_lexer for easy extensibility (iss…

### PR DESCRIPTION
**Pull request by erezsh**
…ue #1329)

### CODE REVIEW

**Comment by erezsh** [comment]
@davidmcnabnz Can you confirm this PR solves your issue?

File: lark/lexer.py [hunk 1]
```diff
@@ -611,8 +622,10 @@ def next_token(self, lex_state: LexerState, parser_state: Any = None) -> Token:
 
 class ContextualLexer(Lexer):
 
-    lexers: Dict[str, BasicLexer]
-    root_lexer: BasicLexer
+    lexers: Dict[str, AbstractBasicLexer]
+    root_lexer: AbstractBasicLexer
+
+    basic_lexer: Type[AbstractBasicLexer] = BasicLexer
```
Comment (MegaIng) [inline lark/lexer.py:628]: Why not `BasicLexer` as the name, following the pattern you using in #1327 ?

Comment (erezsh) [reply lark/lexer.py:629]: Good point.

**Comment by davidmcnabnz** [comment]
Thanks for that @erezsh . Commits look good, and plug up the configurability gap well. I look forward to migrating my lexer/parser over to the new version.

**Pull request merged**